booking/src/main/java/org/dcsa/conformance/standards/booking/action/Carrier_SupplyScenarioParametersAction.java

```diff
@@ -13,15 +13,17 @@ public class Carrier_SupplyScenarioParametersAction extends BookingAction {
   private CarrierScenarioParameters carrierScenarioParameters = null;
 
   private ScenarioType scenarioType;
-  public Carrier_SupplyScenarioParametersAction(String carrierPartyName, @NonNull ScenarioType scenarioType) {
+  public Carrier_SupplyScenarioParametersAction(String carrierPartyName, @NonNull ScenarioType scenarioType, boolean assignCbr) {
     super(carrierPartyName, null, null,
       switch (scenarioType) {
       case REGULAR -> "SupplyCSP";
       case REEFER -> "SupplyCSP-AR";
       case DG -> "SupplyCSP-DG";
     }, -1);
     this.scenarioType = scenarioType;
+    this.assignCbr = assignCbr;
     this.getDspConsumer().accept(getDspSupplier().get().withScenarioType(scenarioType));
+    this.getDspConsumer().accept(getDspSupplier().get().withAssignCbr(assignCbr));
```

Comment:
If the assignCbr stays, then please chain the .with calls instead of doing two lines of .get() + .accept(...)